### DISCUSSION TITLE
Source filters between preview and program scene

### DISCUSSION THREAD

**Discussion by L3Prichaun13** [obsproject/obs-studio#11036] [General]
I myself am a developer and would have no problem adding this functionality to this software along with a toggle setting to allow or disallow it.. but I want to know if it would be acceptable with how the software is expected to work. 
We use sources with filters to overlay messages and scriptures in our videos and use the filter to set the size.. we also use that same source to play videos from the internet and when that happens we need to resize the source using the filter... The problem is that currently, changing the filter settings will update in program scene right away and not just Preview scene first and let me push it to Program when I am ready. 

What I am proposing and I can dev it, is to have a toggle option to modify how filters work so when changing them, they would only update in preview scene until someone pushes it to program scene. 
 This would make our live stream function much smoother, unless I am abusing how the software should be used and there is a better way to handle this..
We have tried setting up multiple sources but then the audio doesn't end up working properly so the filter option would be much simpler

**Comment by mihawk90** [comment]
This is already possible using this toggle:

![image](https://github.com/user-attachments/assets/01f9550d-e4e2-4599-aed1-6732c36fd663)

Do keep in mind that it doesn't work on all sources as per the tooltip.

**Comment by L3Prichaun13** [comment]
I will give this a try next Sunday before. Our Church service... Thank you
very much 😊

**Comment by L3Prichaun13** [comment]
I have the source device set as MiraBox Video Capture.
I checked that box for duplicate source but still when I toggle between the
two filters, it still directly updates Program instead of just Preview.


What I do is toggle the eye icon between the two filters but even with
duplicate source or duplicate scene checked, it's still updating Program...
Is there something else I need to do? Or maybe this is where some
additional programming could come in?

**Reply by mihawk90** [reply]
As the tooltip says it doesn't work with all sources. For USB capture sources it's not possible to duplicate them because the device is already "busy" with the first capture. I'm not sure there's a workaround for that.

I suppose you could open the capture card in a different application that only captures the card and use window capture on that. It's not very clean but it is an option.
